pkg/pluginmanager/manager.go

```diff
@@ -1015,8 +1019,8 @@ func getPluginInfoResource(pluginFilePath string) (*cli.PluginInfo, error) {
 
 // verifyPluginPreDownload verifies that the plugin distribution repo is trusted
 // and returns error if the verification fails.
-func verifyPluginPreDownload(p *discovery.Discovered) error {
-	artifactInfo, err := p.Distribution.DescribeArtifact(p.RecommendedVersion, runtime.GOOS, runtime.GOARCH)
+func verifyPluginPreDownload(p *discovery.Discovered, version string) error {
+	artifactInfo, err := p.Distribution.DescribeArtifact(version, runtime.GOOS, runtime.GOARCH)
```

Comment:
@anujc25 Please pay attention to this change as it is not limited to the new Central Repo feature.
I believe it is safe, but I would appreciate your confirmation.

The reason is:
The verifyPluginPreDownload(..) function now verifies the repo using the requested version of the plugin, instead of always using the recommendedVersion. This should cause not change in functionality IIUC. It is required because with a more targeted query of plugins, the recommendedVersion artifact is not always fetched. For example, if installing an older version of the cluster plugin, say v0.26.0, the "smart" install will only fetch information about that version; so, if we were to attempt to verifyPluginPreDownload() using the RecommendedVersion of v0.28.0, that information would not be available, and would need to be fetched separately from the SQLite DB; so, instead, we just use the requested version.